### PR TITLE
Multi-stage build to cache Apex, conditional dependency management for ARM-systems, ask for CUDA/Apex

### PR DESCRIPTION
**Pull request by MaxiBoether**
- Prevents Apex from being rebuilt everytime the Modyn sourcecode changes
- Finally at least a partial solution for adjusting the environment.yml on mac
- Helps in installing Apex, CUDA, etc. by providing an interactive initial setup script

Partially solves #104.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ services:
 #    volumes:
 #      - .:/modyn_host
 #    shm_size: 4gb
+# CUDASTART (ignore/delete if enabling cuda manually)
```
Comment (vGsteiger) [inline docker-compose.yml:83]: Just from an understanding point, this signs for cuda to include the following code?

File: docker/Dependencies/Dockerfile
Comment (vGsteiger) [inline docker/Dependencies/Dockerfile:1]: Much cleaner!

**Review by vGsteiger — APPROVED** [review]
Much cleaner, thanks!

File: docker-compose.yml (hunk 1 above)
Comment (MaxiBoether) [reply docker-compose.yml:83]: This is for the `initial_setup.sh` script :D To find the lines we are searching for reliably.

File: docker/Dependencies/Dockerfile
Comment (MaxiBoether) [reply docker/Dependencies/Dockerfile:1]: :)

**Pull request merged**